src/vaadin-text-field-mixin.html

```diff
@@ -642,6 +676,9 @@
       this.shadowRoot.querySelector('[name="input"], [name="textarea"]')
         .addEventListener('slotchange', this._onSlotChange.bind(this));
 
+      this._onHelperSlotChange();
+      this.shadowRoot.querySelector('[name="helper"]').addEventListener('slotchange', this._onHelperSlotChange.bind(this));
```

Comment:
BTW we can use `FlattenedNodesObserver` so that we don't need to trigger `this._onHelperSlotChange()` here. Not a blocker.